spmd/compiler/distribute.py

```diff
@@ -384,17 +386,19 @@ def _convert_to_distributed(
         - transformed graph module
         - map from output name to DTensorSpec
     """
+    global logger
+    logger = get_logger("spmd_exp")  # type: ignore
```

Comment:
nit: maybe get the logger in the global scope directly

Reply:
We can't get the logger in global scope because we want torch.distributed to be initialized with PGs instantiated. I have moved printing on a single rank to the logger.